src/tests/Microsoft.PowerFx.Connectors.Tests/OpenApiJsonSerializerTests.cs

```diff
@@ -0,0 +1,144 @@
+            Assert.Equal(@"{""a"":""abc""}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_Bool()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "boolean" }, ["b"] = new OpenApiSchema() { Type = "boolean" } } };
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.New(true)), new NamedValue("b", FormulaValue.New(false)) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal(@"{""a"":true,""b"":false}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_Array()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "array", Format = "int" } } };                            
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.NewSingleColumnTable(FormulaValue.New(1), FormulaValue.New(2))) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal(@"{""a"":[1,2]}", str);
+        }
+
+        [Fact]
+        public void JsonSerializer_Array_Invalid()
+        {
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "array", Format = "int" } } };
+            var record = FormulaValue.NewRecordFromFields(new NamedValue("a", FormulaValue.New(1)), new NamedValue("b", FormulaValue.New("foo")));
+            var table = new InMemoryTableValue(IRContext.NotInSource(TableType.FromRecord(record.Type)), new List<RecordValue>() { record }.Select(r => DValue<RecordValue>.Of(r)));
+            var fields = new List<NamedValue>() { new NamedValue("a", table) };
+
+            var ex = Assert.Throws<ArgumentException>(() => jsonSerializer.Serialize(schema, fields));
+            Assert.Equal("Incompatible Table for supporting array, RecordValue has more than one column - propertyName a, number of fields 2", ex.Message);
+        }
+
+        [Fact]
+        public void JsonSerializer_Date()
+        {
+            var now = DateTime.Now;
+            var jsonSerializer = new OpenApiJsonSerializer();
+            var schema = new OpenApiSchema() { Properties = new Dictionary<string, OpenApiSchema>() { ["a"] = new OpenApiSchema() { Type = "string", Format = "date-time" } } };
+            var fields = new List<NamedValue>() { new NamedValue("a", FormulaValue.New(now)) };
+
+            var str = jsonSerializer.Serialize(schema, fields);
+            Assert.Equal($@"{{""a"":""{now.ToString("o", CultureInfo.InvariantCulture).Replace("+", "\u002B")}""}}", str);
+        }
+    }
```

Comment:
ideas:

- Formula Value has extra property.  (should not be sent) 
- missing / blank property
- nested objects.